### PR TITLE
codec: add desc bytes encoding

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @disksing @coocood  PTAL

### CODE REVIEW

File: src/util/codec/bytes.rs [hunk 1]
```diff
@@ -55,13 +58,26 @@ pub trait BytesEncoder: NumberEncoder {
...
+    } else {
+        bs
+    }
+}
+
 impl<T: Write> BytesEncoder for T {}
 
 // Refer: https://github.com/facebook/mysql-5.6/wiki/MyRocks-record-format#memcomparable-format
-pub fn encode_bytes(key: &[u8]) -> Vec<u8> {
+pub fn encode_bytes(key: &[u8], desc: bool) -> Vec<u8> {
```
Comment (siddontang) [inline src/util/codec/bytes.rs]: Why not supplying another decode_bytes_desc?
Btw, I think using asc as the default argument seems more common.

File: src/util/codec/bytes.rs [hunk 2]
```diff
@@ -70,13 +94,17 @@ pub trait BytesDecoder: NumberDecoder {
     /// Get the remaining length in bytes of current reader.
     fn remaining(&self) -> usize;
 
-    fn decode_bytes(&mut self) -> Result<Vec<u8>> {
+    fn decode_bytes(&mut self, desc: bool) -> Result<Vec<u8>> {
```
Comment (siddontang) [inline src/util/codec/bytes.rs]: why not change decode_bytes at same time?

Comment (BusyJay) [inline src/util/codec/bytes.rs]: Because this method is called infrequently, we don't need another wrapper.

Comment (siddontang) [inline src/util/codec/bytes.rs]: Looks mismatching.
And I see that for number, you have desc decode function too.

Comment (BusyJay) [inline src/util/codec/bytes.rs]: Actually I personally prefer the bool style. Maybe I should create another peer to clean it up.

Comment (siddontang) [inline src/util/codec/bytes.rs]: I like an enum Order {Asc, Desc}.

Comment (siddontang) [inline src/util/codec/bytes.rs]: For a bool, I have to see whether this bool is for ASC or DESC.

Comment (BusyJay) [inline src/util/codec/bytes.rs]: I think it's common to just use a bool to implement this behaviour.

**Comment by ngaut** [comment]
LGTM

**Comment by BusyJay** [comment]
@coocood @disksing PTAL

**Comment by disksing** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**